### PR TITLE
Adding code points and OIDs for Dilithium.

### PR DESCRIPTION
**Pull request by anhu**

### CODE REVIEW

**Comment by anhu** [comment]
@baentsch , can you please merge at your earliest convenience?

File: wireshark/wolfssl-qsc.h [hunk 1]
```diff
@@ -31,6 +37,12 @@
     { 12094, "p256_kyber_90s_level1" }, \
     { 12095, "p384_kyber_90s_level3" }, \
     { 12096, "p521_kyber_90s_level5" }, \
-    { 0xfe0b, "falcon_level1" },\
-    { 0xfe0e, "falcon_level5" },
+    { 0xfe0b, "falcon_level1" }, \
+    { 0xfe0e, "falcon_level5" }, \
+    { 0xfea0, "dilithium_level2" }, \
```
Comment (baentsch) [inline wireshark/wolfssl-qsc.h:42]: SIGs under the "QSC_KEMS" define seem "weird": Is this intentional?

**Review by baentsch — CHANGES_REQUESTED** [review]
> @baentsch , can you please merge at your earliest convenience?

Before doing this, please confirm that your fix should only change the WolfSSL analysis behaviour and not the OQS analysis behaviour (which is default for the Dockerfile). Thus, this fix may not be complete.

File: wireshark/wolfssl-qsc.h (hunk 1 above)
Comment (anhu) [reply wireshark/wolfssl-qsc.h:42]: Yes, I would agree, but this was the convention that you created.  (Falcon was already there in PR 105 which I believe you authored).  Probably better names would be `QSC_CODEPOINTS` and `QSC_OIDS` but that is out of scope of this change.

Comment (anhu) [reply wireshark/wolfssl-qsc.h:42]: Hi @baentsch , 

Sorry for the delayed response on the comment below, had to go eat lunch.  

I prefer you to handle the OQS side.  The reason being you decide your naming conventions and which names are to be displayed or left unknown for your flavour.

**Comment by anhu** [comment]
> Before doing this, please confirm that your fix should only change the WolfSSL analysis...

Confirmed.  I do not want to speak for the OQS Naming conventions and what is left unknown.

**Comment by anhu** [comment]
@baentsch , I have a fix, but it will require some help from you as my `sed` is not as good as yours! 

```
I need the following change added to `epan/dissectors/packet-tls-utils.c` : 
/* RFC 8446 Section 4.2.3 */
const value_string tls13_signature_algorithm[] = {
    { 0x0201, "rsa_pkcs1_sha1" },
    { 0x0203, "ecdsa_sha1" },
    { 0x0401, "rsa_pkcs1_sha256" },
    { 0x0403, "ecdsa_secp256r1_sha256" },
    { 0x0501, "rsa_pkcs1_sha384" },
    { 0x0503, "ecdsa_secp384r1_sha384" },
    { 0x0601, "rsa_pkcs1_sha512" },
    { 0x0603, "ecdsa_secp521r1_sha512" },
    { 0x0804, "rsa_pss_rsae_sha256" },
    { 0x0805, "rsa_pss_rsae_sha384" },
    { 0x0806, "rsa_pss_rsae_sha512" },
    { 0x0807, "ed25519" },
    { 0x0808, "ed448" },
    { 0x0809, "rsa_pss_pss_sha256" },
    { 0x080a, "rsa_pss_pss_sha384" },
    { 0x080b, "rsa_pss_pss_sha512" },
QSC_KEMS
    { 0, NULL }
};
```
I could probably figure it, but its my EOD.  Let me know if you are able to help.

**Comment by anhu** [comment]
To be more explicit, QSC_KEMS needs to be added to tls13_signature_algorithm .

**Comment by anhu** [comment]
The sed wasn't as bad as I thought.  External 3rd party confirmed the fix.

**Comment by baentsch** [comment]
> The sed wasn't as bad as I thought. 

Good to hear that and sorry I currently don't have the bandwidth to set this up myself -- just too many elements involved costing too much time to confirm correct operation :-(

> External 3rd party confirmed the fix.

Which confirmation are you referring to? I only found [this](https://github.com/open-quantum-safe/oqs-demos/issues/164#issuecomment-1330749077) -- and it doesn't look OK. Also, does this confirmation refer to the "wolfssl" flavour or the "OQS" flavour -- which is the default here? I'm still surprised seeing the define `OQS_KEM` used to carry signature names: That does not look right: KEMs and SIGs are clearly separated [in the OQS master file here](https://github.com/open-quantum-safe/openssl/blob/OQS-OpenSSL_1_1_1-stable/qsc.h). I do understand that this approach can work for signatures in the "manipulated" `wolfssl-qsc.h" file but not for the "OQS" version: Hence the question: What confirmation (using which setup) has been given?

**Comment by Jpgig** [comment]
Thanks, Anhu for providing a private git branch path. That worked correctly for me.

**Comment by anhu** [comment]
The private branch @Jpgig is referring to is the branch for this PR: ` wireshark_wolf_dilithium` 

> What confirmation (using which setup) has been given?

This was done using `build.sh` under the `wolfssl` flavour.

**Review by baentsch — APPROVED** [review]

**Comment by baentsch** [comment]
Merging as a stopgap fix for wolfssl. Will follow up with a longterm fix. Will let you know when done @anhu so you could update the wolfssl-specifics if you want.

**Comment by anhu** [comment]
@baentsch , Thank you very much.  I look forward to your update.

**Comment by baentsch** [comment]
@anhu so now done

**Comment by anhu** [comment]
Excellent. Thank you so much @baentsch !!

**Pull request merged**